File: api/airport_manager/src/main/java/com/awbd/airport_manager/service/api/SeatService.java
package com.awbd.airport_manager.service.api;

import com.awbd.airport_manager.dto.SeatDto;
import com.awbd.airport_manager.util.pagination.PagedResponse;
import com.awbd.airport_manager.util.search.dto.SearchDTO;

import java.util.UUID;

public interface SeatService {

    PagedResponse<SeatDto> search(SearchDTO searchDTO);

    SeatDto getById(UUID id);

    SeatDto create(SeatDto dto);

    SeatDto update(UUID id, SeatDto dto);

    void delete(UUID id);
}

File: ui/src/app/core/services/seat.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Seat, SeatForm } from '../../shared/models/seat.model';
import { PagedResponse } from '../../shared/models/paged-response.model';
import { SearchDTO } from '../../shared/models/search.model';

@Injectable({ providedIn: 'root' })
export class SeatService {
  private http = inject(HttpClient);
  private base = `${environment.apiUrl}/api/seats`;

  search(query: SearchDTO): Observable<PagedResponse<Seat>> {
    return this.http.post<PagedResponse<Seat>>(`${this.base}/search`, query);
  }

  getById(id: string): Observable<Seat> {
    return this.http.get<Seat>(`${this.base}/${id}`);
  }

  create(payload: SeatForm): Observable<Seat> {
    return this.http.post<Seat>(this.base, payload);
  }

  update(id: string, payload: SeatForm): Observable<Seat> {
    return this.http.put<Seat>(`${this.base}/${id}`, payload);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
